### PR TITLE
Admin ghost examination window steal/apply

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
Adds the ability for admin ghosts to hold an inventory and to give and take items to other players instantly through the inspect window.

Adds an AdminManager to save the item storage of each admin.
Adds a prefab to be spawned when an admin logs in to act as a holder of the itemstorage.
Adds client server messages to move items on the ghost inventory UI.
PlayerScripts will now have a reference to their connectedPlayer.
PlayerList will now hold a client reference for clients to check if they are an admin.
Fixes a bug with RegisterTile where sameMatrixRelationships and crossMatrixRelationships lists were having its contents removed inside a loop.
Fixes listeners from exosuit, uniform, masks and GasUseSlots itemslots getting the wrong reference once the player swaps to a ghost.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Clothing/StepChanger.cs [hunk 1]
```diff
@@ -58,12 +58,16 @@ private void TryChange(Mind mind, NamedSlot? changeSlot, RegisterPlayer player,
...
 			}
-
 			mind.stepType = (StepType)wearType;
 		}
 	}
 
-	private bool HasHardsuit(Mind mind, NamedSlot? changeSlot) => mind.stepType == StepType.Suit && changeSlot != NamedSlot.outerwear;
+	private bool HasHardsuit(Mind mind, NamedSlot? changeSlot)
+	{
+		if (mind.stepType == StepType.Suit && changeSlot != NamedSlot.outerwear)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Clothing/StepChanger.cs]: what is this change? It was alright and perfectly legible  before

File: UnityProject/Assets/Scripts/Managers/ConnectedPlayer.cs [hunk 2]
```diff
@@ -38,11 +38,15 @@ public GameObject GameObject
 		set
 		{
 			gameObject = value;
+			if (Script)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Managers/ConnectedPlayer.cs:41]: use ``{`` and ``}`` to comply with the if statement code style guidelines.

**Review by corp-0 — CHANGES_REQUESTED** [review]

**Comment by corp-0** [comment]
It looks like you're conflicting with yourself somehow. Make a rebase and accept all your changes, there are files there that haven't been changed in ages so I think it is fine.

File: UnityProject/Assets/Scripts/Managers/ConnectedPlayer.cs (hunk 2 above)
Comment (Aranclanos) [reply UnityProject/Assets/Scripts/Managers/ConnectedPlayer.cs:41]: whats the guideline on 1 line if statements?

**Review by corp-0 — APPROVED** [review]

File: UnityProject/Assets/Scripts/Managers/AdminManager.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	public ItemStorage CreateItemSlotStorage()
+	{
+		var holder = Spawn.ServerPrefab(ItemStorageHolderPrefab, null, gameObject.transform);
+		var itemStorage = holder.GameObject.GetComponent<ItemStorage>();
+		return itemStorage;
+	}
+
+	public void SetLocalAdminGhostStorage()
+	{
+
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Managers/AdminManager.cs]: this is still a wip, right?

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/Managers/AdminManager.cs]: No, just a leftlover

Comment (corp-0) [reply UnityProject/Assets/Scripts/Managers/AdminManager.cs]: can you delete the leftover so we merge this and test on staging?

**Pull request merged**